odoo/addons/base/models/ir_model.py

```diff
@@ -129,7 +129,14 @@ def upsert_en(model, fnames, rows, conflict):
     cols = ", ".join(quote(fname) for fname in fnames)
     values = ", ".join("%s" for row in rows)
     conf = ", ".join(conflict)
-    excluded = ", ".join(f"EXCLUDED.{quote(fname)}" for fname in fnames)
+    excluded = ", ".join(
+        (
+            f"{table}.{quote(fname)} || EXCLUDED.{quote(fname)}"
```

Comment:
`f"COALESCE({table}.{quote(fname)}, '{{}}'::jsonb) || EXCLUDED.{quote(fname)}"`
in case `{table}.{quote(fname)}` was NULL

Reply:
@HydrionBurst thanks, I updated it. Can you merge it?